src/main/java/com/faforever/client/replay/OnlineReplayVaultController.java

```diff
@@ -131,21 +140,36 @@ protected void initSearchController() {
 
-    searchController.addTextFilter("playerStats.player.login", i18n.get("game.player.username"), true);
-    searchController.addTextFilter("mapVersion.map.displayName", i18n.get("game.map.displayName"), false);
-    searchController.addTextFilter("mapVersion.map.author.login", i18n.get("game.map.author"), false);
-    searchController.addTextFilter("name", i18n.get("game.title"), false);
-    searchController.addTextFilter("id", i18n.get("game.id"), true);
-
-    CategoryFilterController featuredModFilterController = searchController.addCategoryFilter("featuredMod.displayName",
+    ReplaySearchPrefs replaySearchPrefs = vaultPrefs.getReplaySearch();
+    TextFilterController textFilterController = searchController.addTextFilter("playerStats.player.login", i18n.get("game.player.username"), true);
+    textFilterController.setText(replaySearchPrefs.getPlayerNameField());
+    replaySearchPrefs.playerNameFieldProperty().bind(textFilterController.textFieldProperty().when(showing));
+    textFilterController = searchController.addTextFilter("mapVersion.map.displayName", i18n.get("game.map.displayName"), false);
+    textFilterController.setText(replaySearchPrefs.getMapNameField());
+    replaySearchPrefs.mapNameFieldProperty().bind(textFilterController.textFieldProperty().when(showing));
+    textFilterController = searchController.addTextFilter("mapVersion.map.author.login", i18n.get("game.map.author"), false);
+    textFilterController.setText(replaySearchPrefs.getMapAuthorField());
+    replaySearchPrefs.mapAuthorFieldProperty().bind(textFilterController.textFieldProperty().when(showing));
+    textFilterController = searchController.addTextFilter("name", i18n.get("game.title"), false);
+    textFilterController.setText(replaySearchPrefs.getTitleField());
+    replaySearchPrefs.titleFieldProperty().bind(textFilterController.textFieldProperty().when(showing));
+    textFilterController = searchController.addTextFilter("id", i18n.get("game.id"), true);
+    textFilterController.setText(replaySearchPrefs.getReplayIDField());
+    replaySearchPrefs.replayIDFieldProperty().bind(textFilterController.textFieldProperty().when(showing));
+
+    featuredModFilterController = searchController.addCategoryFilter("featuredMod.displayName",
         i18n.get("featuredMod.displayName"), List.of());
 
     featuredModService.getFeaturedMods().map(FeaturedMod::displayName)
                       .collectList()
                       .publishOn(fxApplicationThreadExecutor.asScheduler())
-                      .subscribe(featuredModFilterController::setItems);
+                      .subscribe((mods) -> {
+                        featuredModFilterController.setItems(mods);
+                        replaySearchPrefs.featuredModFilterProperty().forEach((item) -> featuredModFilterController.checkItem(item));
+                        Bindings.bindContent(replaySearchPrefs.featuredModFilterProperty(), featuredModFilterController.getCheckedItems());
```

Comment:
I would do the actual binding of the content in the onShow method as well to ensure if it is made visible and invisible the binding will still be there

Reply:
If we do this then there is a race condition where the binding will sometimes get set to the empty controller items before checked items getting set here, therefore overwriting the preferences.

Reply:
We could create a copy of the list during initialization and use that in the subscribe.

Reply:
I dont love this

Reply:
Well as it is right now there are cases where the selection would not be saved to the prefs